admin/class-yoast-notification-center.php

```diff
@@ -54,74 +57,162 @@ public static function get() {
+		$notification    = $notification_center->get_notification_by_id( $notification_id );
+
+		if ( false === ( $notification instanceof Yoast_Notification ) ) {
+			die( '-1' );
+		}
+
+		if ( $notification_center->maybe_dismiss_notification( $notification ) ) {
+			die( '1' );
+		}
+
+		die( '-1' );
 	}
 
 	/**
-	 * Clear local stored notifications
+	 * Check if the user has dismissed a notification
+	 *
+	 * @param string   $dismissal_key The dismissal key.
+	 * @param null|int $user_id       User ID to check on.
+	 *
+	 * @return bool
 	 */
-	private function clear_notifications() {
-		$this->notifications = array();
+	public static function is_notification_dismissed( $dismissal_key, $user_id = null ) {
+		// Empty dismissal keys can never be set i.e. non-persistent notifications are always active.
+		if ( empty( $dismissal_key ) ) {
+			return false;
+		}
+
+		$user_id = ( ! is_null( $user_id ) ? $user_id : get_current_user_id() );
+
+		$current_value = get_user_meta( $user_id, $dismissal_key, $single = true );
+
+		return ! empty( $current_value );
 	}
 
 	/**
-	 * Remove transient when the plugin is deactivated
+	 * Check if the nofitication is being dismissed
+	 *
+	 * @param string|Yoast_Notification $notification Notification to check dismissal of.
+	 * @param string                    $meta_value   Value to set the meta value to if dismissed.
+	 *
+	 * @return bool True if dismissed.
 	 */
-	public function deactivate_hook() {
-		$this->clear_notifications();
+	public static function maybe_dismiss_notification( $notification, $meta_value = 'seen' ) {
+		$notification_id = self::get_notification_id( $notification );
+		$dismissal_key   = self::get_dismissal_key( $notification );
+
+		if ( false === $notification_id || false === $dismissal_key ) {
+			return false;
+		}
+
+		// If notification is already dismissed, we're done.
+		if ( self::is_notification_dismissed( $dismissal_key ) ) {
+			return true;
+		}
+
+		$is_dismissing = $dismissal_key === self::get_user_input( 'notification' );
+		$is_dismissing = $is_dismissing || ( $notification_id === self::get_user_input( 'notification' ) );
+
+		$user_nonce = self::get_user_input( 'nonce' );
+
+		// Can be dismissed by dismissal_key or notification_id.
+		if ( ! $is_dismissing ) {
+			return false;
+		}
+
+		if ( false === wp_verify_nonce( $user_nonce, $notification_id ) ) {
+			return false;
+		}
+
+		// Dismiss notification.
+		update_user_meta( get_current_user_id(), $dismissal_key, $meta_value );
+
+		return true;
 	}
 
 	/**
-	 * Write the notifications to a cookie (hooked on shutdown)
+	 * Clear dismissal information for the specified Notification
+	 *
+	 * When a cause is resolved, the next time it is present we want to show
+	 * the message again.
+	 *
+	 * @param string|Yoast_Notification $notification Notification to clear the dismissal of.
+	 *
+	 * @return bool
 	 */
-	public function set_transient() {
+	public function clear_dismissal( $notification ) {
```

Comment:
But that's not what it does. It removes the dismissal flag for the notification, making it show up again when the condition is met in the future.